source/solvers/heat_transfer.cc

```diff
@@ -969,7 +1019,780 @@ HeatTransfer<dim>::solve_linear_system(const bool initial_step,
+    Utilities::MPI::max(maximum_temperature, mpi_communicator);
+
+  volume_integral = Utilities::MPI::sum(volume_integral, mpi_communicator);
+  temperature_integral =
+    Utilities::MPI::sum(temperature_integral, mpi_communicator);
+  double temperature_average = temperature_integral / volume_integral;
+
+  // Calculate standard deviation
+  double temperature_variance_integral = 0;
+  for (const auto &cell : this->dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned())
+        {
+          fe_values_ht.reinit(cell);
+          fe_values_ht.get_function_values(this->present_solution,
+                                           local_temperature_values);
+
+          if (gather_vof)
+            {
+              // Get VOF active cell iterator
+              typename DoFHandler<dim>::active_cell_iterator cell_vof(
+                &(*(this->triangulation)),
+                cell->level(),
+                cell->index(),
+                dof_handler_vof);
+
+              // Gather VOF information
+              fe_values_vof->reinit(cell_vof);
+              fe_values_vof->get_function_values(
+                *this->multiphysics->get_solution(PhysicsID::VOF),
+                phase_values);
+            }
+
+          for (unsigned int q = 0; q < n_q_points; q++)
+            {
+              switch (monitored_fluid)
+                {
+                  case Parameters::FluidIndicator::both:
+                    {
+                      phase_coefficient = 1.;
+                      break;
+                    }
+                  case Parameters::FluidIndicator::fluid0:
+                    {
+                      if (gather_vof)
+                        phase_coefficient = 1. - phase_values[q];
+                      else
+                        {
+                          // In the case of monophase simulations, "fluid0" is
+                          // equivalent to "both" (all domain) calculation
+                          phase_coefficient = 1.;
+                        }
+                      break;
+                    }
+                  case Parameters::FluidIndicator::fluid1:
+                    {
+                      if (gather_vof)
+                        phase_coefficient = phase_values[q];
+                      else
+                        {
+                          // Defensive only here, this has been checked already
+                          // in simulation_parameters.h
+                          throw std::logic_error(
+                            "Inconsistency in .prm!\n when VOF = false"
+                            "\n use (default value): set postprocessed fluid = both"
+                            "\n or: set postprocessed fluid = fluid 0");
+                        }
+                      break;
+                    }
+                  default:
+                    throw std::runtime_error(
+                      "Unsupported number of fluids (>2)");
```

Comment:
Do you think this phase coefficient setting could be moved to a function? I feel like this block of code and the one above are very similar in spirit (or maybe I am wrong?)

Reply:
no you are totally right! Very good suggestion indeed